File: src/main/java/com/example/myapplication/Singleton.java
package com.example.myapplication;

public class Singleton {
    private static Singleton instance = null;
private  Singleton(){

}
private synchronized static void  createInstance(){
    if(instance==null){
        instance=new Singleton();
    }
}
public static Singleton getInstance(){
    if (instance==null) createInstance():
    return instance
}
}

File: src/main/java/com/example/myapplication/Singleton.kt
package com.example.myapplication

package com.example.myapplication
object Singleton {
    private var instance: Singleton? = null
    @kotlin.jvm.Synchronized
    private fun createInstance() {
        if (instance == null) {
            instance = Singleton()
        }
    }

    fun getInstance(): Singleton? {
        if (instance == null) createInstance()
        return instance
    }
}